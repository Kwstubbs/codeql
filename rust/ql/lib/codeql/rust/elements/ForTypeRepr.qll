// generated by codegen, do not edit
/**
 * This module provides the public class `ForTypeRepr`.
 */

private import internal.ForTypeReprImpl
import codeql.rust.elements.ForBinder
import codeql.rust.elements.TypeRepr

/**
 * A type with a higher-ranked `for` modifier. This is currently not valid Rust syntax (`for<...>` can
 * only be applied to traits to form a `TypeBound`).
 *
 * For example:
 * ```rust
 * fn foo(value: for<'a> usize) {}  // DOESN'T COMPILE
 * //            ^^^^^^^^^^^^^
 * ```
 */
final class ForTypeRepr = Impl::ForTypeRepr;
