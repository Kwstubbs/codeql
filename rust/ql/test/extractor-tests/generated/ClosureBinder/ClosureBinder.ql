// generated by codegen, do not edit
import codeql.rust.elements
import TestUtils

query predicate instances(
  ClosureBinder x, string hasGenericParamList__label, string hasGenericParamList
) {
  toBeTested(x) and
  not x.isUnknown() and
  hasGenericParamList__label = "hasGenericParamList:" and
  if x.hasGenericParamList() then hasGenericParamList = "yes" else hasGenericParamList = "no"
}

query predicate getGenericParamList(ClosureBinder x, GenericParamList getGenericParamList) {
  toBeTested(x) and not x.isUnknown() and getGenericParamList = x.getGenericParamList()
}
