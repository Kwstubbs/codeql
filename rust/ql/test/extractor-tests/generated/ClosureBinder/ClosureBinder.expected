instances
| gen_closure_binder.rs:7:21:7:43 | ClosureBinder | hasGenericParamList: | yes |
getGenericParamList
| gen_closure_binder.rs:7:21:7:43 | ClosureBinder | gen_closure_binder.rs:7:24:7:43 | <...> |
